internal/compile.bzl

```diff
@@ -337,14 +341,23 @@ def proto_compile(ctx, options, extra_protoc_args, extra_protoc_files):
             for f in plugin_outputs:
                 print("EXPECTED OUTPUT:", f.path)  # buildifier: disable=print
 
-        # Run protoc
+        env = {}
+        if (len(env) > 0) and plugin.use_built_in_shell_environment:
```

Comment:
Is it actually forbidden to use `env` at the same time as `use_default_shell_env` at the Bazel level? The docs make no indication that these are mutually exclusive and it may be useful to append env to a cleared default env if possible.

Reply:
I can't find the comment where I read about this restriction, but it seems to be enforced in the code here:
https://github.com/bazelbuild/bazel/blob/2f8b4849bb4732c52eef0f194cf21ccf89811132/src/main/java/com/google/devtools/build/lib/analysis/actions/SpawnAction.java#L1055...

That builder is used here, in the run_shell implementation:
https://github.com/bazelbuild/bazel/blob/2f8b4849bb4732c52eef0f194cf21ccf89811132/src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkActionFactory.java#L593

Reply:
Aha, a related issue: https://github.com/bazelbuild/bazel/issues/5980